rocAL/source/api/rocal_api_meta_data.cpp

```diff
@@ -51,12 +51,14 @@ void
 
 RocalMetaData
     ROCAL_API_CALL
-    rocalCreateLabelReader(RocalContext p_context, const char* source_path) {
+    rocalCreateLabelReader(RocalContext p_context, const char* source_path, const char* file_list_path) {
     if (!p_context)
         THROW("Invalid rocal context passed to rocalCreateLabelReader")
     auto context = static_cast<Context*>(p_context);
-
-    return context->master_graph->create_label_reader(source_path, MetaDataReaderType::FOLDER_BASED_LABEL_READER);
+    if (strlen(file_list_path) == 0)
```

Comment:
is this for checking empty string. please use is_empty() instead

Reply:
The source path is passed as const char array so strlen is used here
We are creating different metadata readers depending on whether file_list_path is passed or not so both are needed.

Reply:
In that case, check if file_list_path == nullptr ?

Reply:
Both source_path and file_list_path have default values of "" so nullptr checks won't work for them